### PR TITLE
DSP_DSP_Add_Some_Functions

### PR DESCRIPTION
**Pull request by Jhon5912**
Hopefully done it wright this time with pushing files to main Citra without merging other files with it..

Just to add some more DSP functions, Fixes Unknown ?..

Reviews welcome ?,  And little boost of help at first..

These Buffer Functions are Recorded from citra_log.txt with wwylele teakra branch.. With a Certain Game.

        buffer[4] = 4;
        buffer[20] = 20;

To add are not to add ^ ?

My deepest apologies for some mistakes I'v made in the pass, Turns out I need to use main master Citra branch for these new merges so no other files are branched with it.

Hope to add some more Contributions soon.

Thanks all.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4361)

### CODE REVIEW

**Comment by Jhon5912** [comment]
Hum how one fix travis one issue..

**Comment by RocketRobz** [comment]
`/Users/travis/build/citra-emu/citra/src/core/hle/service/dsp/dsp_dsp.cpp:377:89: error: call to unavailable member function 'value': introduced in macOS 10.14
    const PAddr physical_address = Memory::TryVirtualToPhysicalAddress(virtual_address).value();

/Users/travis/build/citra-emu/citra/src/core/hle/service/dsp/dsp_dsp.cpp:390:86: error: call to unavailable member function 'value': introduced in macOS 10.14
    const VAddr virtual_address = Memory::PhysicalToVirtualAddress(physical_address).value();`

It appears the use of `.value()` is causing the error.

**Comment by Jhon5912** [comment]
No solution found on my end..

**Comment by Jhon5912** [comment]
Edited but not sure ^ ?

File: src/core/hle/service/dsp/dsp_dsp.cpp [hunk 1]
```diff
@@ -104,10 +178,8 @@ void DSP_DSP::WriteProcessPipe(Kernel::HLERequestContext& ctx) {
         break;
     case DspPipe::Binary:
         ASSERT(buffer.size() >= 8);
-        buffer[4] = 1;
-        buffer[5] = 0;
-        buffer[6] = 0;
-        buffer[7] = 0;
+        buffer[4] = 4;
+        buffer[20] = 20;
```
Comment (FearlessTobi) [inline src/core/hle/service/dsp/dsp_dsp.cpp]: What are those two magic numbers?

Comment (Jhon5912) [reply src/core/hle/service/dsp/dsp_dsp.cpp]: @wwylele teakra branch came up with it for pm x/y buffer seems to match the buffer number, 20 also seam to bypass a certain freeze he may have in his branch.

Are maybe edit it back to its original for now ?

Comment (wwylele) [reply src/core/hle/service/dsp/dsp_dsp.cpp]: To clarify, I have no idea what these numbers are and I haven't put them into any of my code. Stop tagging me as I am not responsible for your code.

Comment (Jhon5912) [reply src/core/hle/service/dsp/dsp_dsp.cpp]: I'm not at wwylele its you're build
https://ci.appveyor.com/project/wwylele/citra/build/1.0.870/job/0i8x38pppudsqprk/artifacts

error crash in pokemon x/y
[  48.111803] HW.Memory <Error> core\memory.cpp:Memory::GetPhysicalPointer:287: unknown GetPhysicalPointer @ 0x00010560

WriteProcessPipe

Service.DSP <Warning> core\hle\service\dsp\dsp_dsp.cpp:Service::DSP::DSP_DSP::WriteProcessPipe:372: channel=2, size=0x4, buffer_size=4

= buffer[4] = 4; ?

[  48.034467] Service.DSP <Warning> core\hle\service\dsp\dsp_dsp.cpp:Service::DSP::DSP_DSP::WriteProcessPipe:372: channel=3, size=0x20, buffer_size=20
[  48.072721] Service.DSP <Warning> core\hle\service\dsp\dsp_dsp.cpp:Service::DSP::DSP_DSP::ProcessPipeEvent:283: Pipe signal received for pipe 3
[  48.072732] Service.DSP <Warning> core\hle\service\dsp\dsp_dsp.cpp:Service::DSP::DSP_DSP::ProcessPipeEvent:287: Readable size = 32
[  48.073629] Service.DSP <Warning> core\hle\service\dsp\dsp_dsp.cpp:Service::DSP::DSP_DSP::ReadPipeIfPossible:456: channel=3, peer=0, size=0x0020, pipe_readable_size=0x0020

= buffer[20] = 20; ? This dose fixe the PM X/Y freeze above

Comment (Jhon5912) [reply src/core/hle/service/dsp/dsp_dsp.cpp]: Full Log
http://s000.tinyupload.com/download.php?file_id=03200554849382053145&t=0320055484938205314545598

**Comment by Jhon5912** [comment]
Edited back to some original code.

**Comment by FearlessTobi** [comment]
Closing cause the creator was suspended.

**Pull request closed without merging**